utopia-remix/app/routes/p.$id.tsx

```diff
@@ -0,0 +1,30 @@
+import type { LoaderFunctionArgs } from '@remix-run/node'
+import { validateProjectAccess } from '../handlers/validators'
+import { proxy } from '../util/proxy.server'
+import { UserProjectPermission } from '~/types'
+import { type Params, redirect } from '@remix-run/react'
+import { getProjectIdFromParams } from '../util/api.server'
+
+export async function loader(args: LoaderFunctionArgs) {
+  return getProjectForEditor(args.request, args.params)
+}
+
+const validator = validateProjectAccess(UserProjectPermission.CAN_VIEW_PROJECT, {
+  canRequestAccess: true,
+  getProjectId: (params) => getProjectIdFromParams(params, 'id'),
+})
+
+export async function getProjectForEditor(req: Request, params: Params<string>) {
+  const validatorResult = await validator(req, params)
+  if (validatorResult.ok) {
+    const proxyResponse: Response = (await proxy(req, {
+      rawOutput: true,
+    })) as Response
+    return new Response(proxyResponse.body, {
+      headers: { 'content-type': 'text/html' },
```

Comment:
it would be good to return all response headers here (except `host`, see the implementation in `proxy`), maybe overriding the content type. But isn't the content type already returned with the response headers?

Reply:
I actually tried returning the `proxyResponse` itself but it didn't work. I'll try to maybe just use `handleMethod` logic here and see if I can unify both